integration_test/sql/sandbox.exs

```diff
@@ -189,4 +202,45 @@ defmodule Ecto.Integration.SandboxTest do
       end)
     end
   end
+
+  describe "start_owner!/2" do
+    test "checks out the connection" do
+      assert_raise DBConnection.OwnershipError, ~r"cannot find ownership process", fn ->
+        TestRepo.all(Post)
+      end
+
+      owner = Sandbox.start_owner!(TestRepo)
+      assert TestRepo.all(Post) == []
+
+      :ok = Sandbox.stop_owner(owner)
+      message = "owner #{inspect(owner)} exited"
+
+      try do
+        TestRepo.all(Post) == []
+      catch
+        :exit, {{:shutdown, ^message}, _} ->
+          :ok
```

Comment:
Use `assert {{:shutdown, ^message}, _} = catch_exit(TestRepo.all(Post))`?

Reply:
Actually, it seems this can fail with other reasons... so maybe `assert catch_exit(...)` is enough?

Reply:
I run:

```
      message = "owner #{inspect(owner)} exited"
       assert {{:shutdown, ^message}, _} = catch_exit(TestRepo.all(Post))
```

100 times and it was stable and while it could have failed on 101st time, I'd keep it as is for now, and I'll make it looser when it fails, wdyt?

Reply:
It already failed on CI. :)